docs/tokens.md

```diff
@@ -0,0 +1,257 @@
+```
+
+to add more make it into an array, e.g.:
+
+```toml
+unconditionally_valid_tokens=["0x0000000000000000000000000000000000000000","0x186E485150dDAB799Be8e25f87D717dEc45a86e5"]
+```
+
+### **Tokens not available for fee payment**
+
+In some operations (Withdraw, ForcedExit, Transfer) the fee token cannot be selected and must be paid using the token
+being transferred or withdrawn. However, if the transaction involves a token without a liquid market price, then (as per
+RIF Rollup documentation) fees for these operations are supposed to be paid using a different token but unfortunately,
+this is currently not implemented.
+
+Trying to transfer or withdraw a token that doesn’t have a liquid market price will **fail**. There is a
+[workaround](https://discord.com/channels/722409280497516566/829104124145696808/920231908460335114) though. To pay the
+fee in a different token, you need to create a batch of transactions (there is a convenience class for it in the
+[JS SDK](https://docs.zksync.io/api/sdk/js/accounts/#batch-builder)). The first transaction should be of the token you
+wish to transfer/withdraw. Then since we’re assuming a scenario where no other tx needs to be sent, the user must
+include a dummy transaction whose sole purpose is to pay for the fees. This second transaction may be of the same type
+(i.e. transfer/withdraw) but for one of the available token fees. Since it’s a dummy transaction the user will set the
+amount (transferred or withdrawn) as 0, and she must include a fee large enough (in this case approximately twice the
+size) to cover the cost for both transactions. While this method works, the
+[Transaction Batching](#transaction-batching) section presents a more user-friendly approach.
+
+## **Fees**
+
+Fees for each type of transaction are computed based on three main factors: Amount of data that will be sent to the L1
+network. Current gas price on L1 (e.g. currently around 0.06 gwei on RSK). Cost of computational resources to generate a
+proof for a block including the transaction.
+
+Fees are low since these fees are distributed among all the transactions included in the rollup block. The minimum fee
+required to process a transaction can be queried using the `get_tx_fee` API entry
+([v0.1](https://zksync.io/api/v0.1.html#get-tx-fee) or
+[v0.2](https://docs.zksync.io/apiv02-docs/#fees-api-v0.2-fee-post))
+
+### **Fee costs**
+
+In RIF Rollup the cost of every transaction has two components:
+
+- **Off-chain part (storage + prover costs):** the cost of the state storage and the SNARK (zero-knowledge proof)
+  generation. This part depends on the use of hardware resources and is therefore invariable. Our benchmarks give
+  estimates of ~0.001 USD per transfer.
+- **On-chain part (gas costs):** for every RIF Rollup block, the validator must pay Rootstock gas to verify the SNARK,
+  plus additionally ~0.4k gas per transaction to publish the state 𝛥. The on-chain part is a variable that depends on
```

Comment:
I would point directly to the code where this is hard-code, so we change the code we don't need to update the documentation.

Reply:
I have no idea where this value is hardcoded tbh.

Reply:
here is the harcoded value. 
https://github.com/rsksmart/rif-rollup/blob/ROLLUP-197/erc20docs/core/bin/zksync_api/src/fee_ticker/mod.rs#L269

Reply:
thanks @franciscotobar, but that just seems to be -1000 (`Ratio::from_integer(BigUint::from(10u32).pow(3u32)).inv()`) instead of the `~0.4k` that @antomor pointed out?
TBF I don't know what those values are either...

Reply:
added the link to the 0.001 value and reworded the on-chain part, removing the currently misleading 0.4k value